actions.go

```diff
@@ -0,0 +1,221 @@
+	Name        string      `json:"name"`
+	Description *string     `json:"description,omitempty"`
+	Owner       graphql.ID  `json:"ownerId"`
+	Action      *graphql.ID `json:"actionId,omitempty"`
+	Filter      *graphql.ID `json:"filterId,omitempty"`
+	// This is being explictly left out to reduce the complexity of the implementation
+	// action *CustomActionsWebhookActionCreateInput
+}
+
+func (s *CustomActionsWebhookAction) Headers() map[string]string {
+	return CustomActionsToHeaders(s.HeadersString)
+}
+
+func CustomActionsToHeaders(input string) map[string]string {
+	output := map[string]string{}
+	headers := strings.Split(input, "\n")
+	for _, header := range headers {
+		parts := strings.Split(header, "=")
+		// TODO: handle parts split error / len != 2
+		s, _ := strconv.Unquote(parts[1])
+		// TODO: handle Unquote error
+		output[parts[0]] = s
+	}
+	return output
+}
+
+func CustomActionsFromHeaders(input map[string]string) string {
+	// To ensure consistent output we use a sorted list of keys
+	// This way the string of headers always appears in alphabetical order
+	keys := make([]string, 0, len(input))
+	for k := range input {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	builder := strings.Builder{}
+	for _, k := range keys {
+		builder.WriteString(fmt.Sprintf("%s=\"%s\"\n", k, input[k]))
+	}
+	return strings.TrimSuffix(builder.String(), "\n")
+}
+
+func (client *Client) CreateWebhookAction(input CustomActionsWebhookActionCreateInput) (*CustomActionsExternalAction, error) {
+	var m struct {
+		Payload struct {
+			WebhookAction CustomActionsExternalAction
+			Errors        []OpsLevelErrors
+		} `graphql:"customActionsWebhookActionCreate(input: $input)"`
+	}
+	marshalledHeaders := CustomActionsFromHeaders(input.Headers)
+	input.HeadersString = &marshalledHeaders
+	v := PayloadVariables{
+		"input": input,
+	}
+	if err := client.Mutate(&m, v); err != nil {
+		return nil, err
+	}
+	return &m.Payload.WebhookAction, FormatErrors(m.Payload.Errors)
+}
+
+// TODO: Not implemented in the API yet
+//func (client *Client) GetCustomAction(input IdentifierInput) (*CustomActionsTriggerDefinition, error) {
+
+//}
+
+func (client *Client) ListCustomActions(variables *PayloadVariables) (CustomActionsExternalActionsConnection, error) {
+	var q struct {
+		Account struct {
+			Actions CustomActionsExternalActionsConnection `graphql:"customActionsExternalActions(after: $after, first: $first)"`
+		}
+	}
+	if variables == nil {
+		variables = &PayloadVariables{
+			"after": "",
+			"first": client.pageSize,
+		}
+	}
+	if err := client.Query(&q, nil); err != nil {
+		return CustomActionsExternalActionsConnection{}, err
+	}
+	for q.Account.Actions.PageInfo.HasNextPage {
+		(*variables)["after"] = q.Account.Actions.PageInfo.End
+		resp, err := client.ListCustomActions(variables)
+		if err != nil {
+			return CustomActionsExternalActionsConnection{}, err
+		}
+		q.Account.Actions.Nodes = append(q.Account.Actions.Nodes, resp.Nodes...)
```

Comment:
i'm not too thrilled about this section but i'm not sure of a better way to do it